### DISCUSSION TITLE
Open question: HNSW vs exhaustive KNN when number of chunks is around 10k

### DISCUSSION THREAD

**Discussion by cherifbenham** [Azure-Samples/chat-with-your-data-solution-accelerator#958] [Q&A]
Hello friends
Does somebody know if there is a significant gain in accuracy/recall when using exhaustive knn over hnws?
I guess the search latency/speed is still reasonable for n_chunks around 10k
What are your thoughts on this?

Thanks !!!!

**Comment by ross-p-smith** [comment]
I would say that this depends on the data/index you are using. I would look at something like the https://github.com/microsoft/rag-experiment-accelerator to look at the gains and when you have finished then configure this repository based on your answer
